awscli/customizations/cloudformation/yamlhelper.py

```diff
@@ -96,8 +96,17 @@ def yaml_parse(yamlstr):
         loader.add_constructor(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, 
                                _dict_constructor)
         loader.add_multi_constructor("!", intrinsics_multi_constructor)
+
         return yaml.load(yamlstr, loader)
 
+def yaml_compose(yamlstr):
+    """Returns a tree after parsing the string"""
+
+    loader = SafeLoaderWrapper
+    loader.add_constructor(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, 
+                           _dict_constructor)
+    loader.add_multi_constructor("!", intrinsics_multi_constructor)
+    return yaml.compose(yamlstr, loader)
```

Comment:
This will fail on JSON files.  While YAML can parse JSON there are some limitations that require the JSON parsers.

Reply:
Just for compose, or for yaml_parse as well?